cachito/workers/pkg_managers/general_js.py

```diff
@@ -345,6 +355,8 @@ def upload_non_registry_dependency(dep_identifier, version_suffix):
     """
     with tempfile.TemporaryDirectory(prefix="cachito-") as temp_dir:
         env = {
+            # This is set since the home directory must be writable for npm
+            "HOME": temp_dir,
```

Comment:
Why does it need access to the HOME dir?

Reply:
I added a comment explaining this and also changed this to inherit the home directory instead.